### DISCUSSION TITLE
How to use bytenode about this ?

### DISCUSSION THREAD

**Discussion by allran** [cawa-93/vite-electron-builder#216] [Q&A]
how to use [bytenode](https://github.com/bytenode/bytenode) to set and load this?

**Comment by cawa-93** [comment]
I looked through the documentation. It doesn't look like any special settings are needed. You can use this tool as usual.

**Reply by Nisthar** [reply]
> I looked through the documentation. It doesn't look like any special settings are needed. You can use this tool as usual.

What do you mean no special settings are needed? is it included in the template?

**Reply by cawa-93** [reply]
I mean, you can use it the same way like in any other project. This can be a plugin for rollup, or a command in the npm script, or something else.

**Reply by Nisthar** [reply]
@cawa-93 i gave it a try using the command `bytenode -c packages/main/dist/index.cjs`. it creates a compiled file `packages/main/dist/index..jsc`. If i try to run it, it gives the error 
```
Error: Invalid or incompatible cached data (cachedDataRejected)
at Object.Module._extensions.<computed> [as .jsc] (D:\node_projects\myproject\node_modules\[4mbytenode[24m\index.js:273:11)
```

**Reply by Nisthar** [reply]
i also tried with the output file generated from watch command (index.js) it still gives the same error.

**Reply by cawa-93** [reply]
@Nisthar I don't know what the problem might be.
Maybe it will help you:
- https://github.com/bytenode/bytenode/issues/63#issuecomment-652866279
- https://github.com/bytenode/bytenode/tree/master/examples/electron-hello-world

If you find a solution - describe it here for posterity.

**Reply by Nisthar** [reply]
@cawa-93 oh it seems the error was because of nodejs. I had to execute it using electron.

**Reply by Nisthar** [reply]
@cawa-93 sry still can't get it to work.
This is my code:
`packages/main/src/index.js`
```js
"use strict";

const bytenode = require("bytenode");
const fs = require("fs");
const v8 = require("v8");
const path = require("path");

v8.setFlagsFromString("--no-lazy");

(async () => {
  try {
    if (!fs.existsSync(path.join(__dirname, "./../dist/main.jsc"))) {
      await bytenode.compileFile(
        path.join(__dirname, "main.src.js"),
        path.join(__dirname, "./../dist/main.jsc")
      );
    }
    console.log(path.join(__dirname, "main.src.js"));
    require(path.join(__dirname, "main.jsc"));
  } catch (e) {
    console.log(e);
  }
})();
```
main.src.js:
```js
const { app, BrowserWindow } = require("electron");
const { join } = require("path");
const { URL } = require("url");
...everything else is the same
```

**Reply by SiriffoS** [reply]
Hey! Did anyone of you find a solution? Im trying to implement bytenode and cant make it work.

**Reply by adnanlah** [reply]
Hello, anyone implemented bytenode successfully before?
